dashboard/test/ui/features/star_labs/applab/shared_apps.feature

```diff
@@ -5,6 +5,8 @@ Feature: App Lab Scenarios
   Background:
     Given I start a new Applab project
     And I wait for the page to fully load
+    # Make sure we can navigate to another page without getting an alert
+    And I disable onBeforeUnload
```

Comment:
Do you mind explaining why we're getting an alert without including this line?

Reply:
Welp I hit so many of these across many tests I ended up deciding to disable this popup in UI tests. This step just runs some code that sets a flag:

https://github.com/code-dot-org/code-dot-org/blob/67907f0f23f60705d0ba44d60def05231f4187bf/dashboard/test/ui/features/step_definitions/steps.rb#L1226-L1228

Which then gets used to disable the event handler logic.

I ended up setting this flag to true by default in tests (which made the UI test step to set it to true unnecessary). The downside here is that UI tests wouldn't catch if this popup was appearing unexpectedly, but I think it is pretty annoying to have to disable it in any Applab test, and we do have unit tests that test the core behavior. It also does not disable the new change handler that marks a project as having changed on any change in the editor (ie, `studioApp().addChangeHandler(project.projectChanged);`), and no tests failed.

Reply:
Curious if you have thoughts about whether disabling these popups in UI tests is a bad idea though Alice!

Reply:
I pulled the latest version of your branch, and for me, it seems like the auto-saving is not working locally. I think it's because `ignoreOnBeforeUnload` is set to `true` by default. It looks like `window.__TestInterface` is defined not just for UI tests but in general when the app is loaded. When I set it to `false` the autosave works as expected.

It looks like there's been online discussion on how to disable this `onbeforeunload` popup in UI tests so it seems like it's okay to disable. I agree it's annoying to have to disable it in any Applab test.

I thought maybe for the UI tests, we could run/reset after a project change, but this also doesn't prevent the popup window, I think because `hasProjectChanged` is still set to `true`. Maybe you could add an update so that when a program is run/reset, the `projectHasChanged` property is set to `false`? Then we could just add a step in UI tests to run/reset the program to avoid the popup. 

This would also mean that user would not be presented with the pop-up unnecessarily if the project is already saved after run/reset and no change in the workspace has been detected.